linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/grpc/GrpcClassifier.scala

```diff
@@ -0,0 +1,94 @@
+package io.buoyant.linkerd.protocol.h2.grpc
+
+import com.twitter.finagle.buoyant.h2.service.{H2Classifier, H2ReqRep, H2ReqRepFrame}
+import com.twitter.finagle.service.ResponseClass
+import com.twitter.util.Return
+import io.buoyant.grpc.runtime.GrpcStatus
+import io.buoyant.grpc.runtime.GrpcStatus.{Ok, Unavailable}
+
+
+trait GrpcClassifier extends H2Classifier {
+  /**
+    * @inheritdoc
+    * Since GRPC sends status codes in the
+    * [[com.twitter.finagle.buoyant.h2.Frame.Trailers Trailers]] frame of an H2
+    * stream, we can never attempt early classification
+    */
+  override val responseClassifier: PartialFunction[H2ReqRep, ResponseClass] =
+    PartialFunction.empty
+}
+
+
+/**
+  * [[H2Classifier]]s for gRPC
+  */
+object GrpcClassifiers {
+
+  /**
+    * [[H2Classifier]] that classifies all error status
+    * codes as [[ResponseClass.RetryableFailure]]
+    */
+  object AlwaysRetryable extends GrpcClassifier {
+
+    override val streamClassifier: PartialFunction[H2ReqRepFrame, ResponseClass] = {
+      case H2ReqRepFrame(_, Return((_, Some(Return(GrpcStatus(Ok(_))))))) =>
+        ResponseClass.Success
+      case H2ReqRepFrame(_, Return((_, Some(Return(GrpcStatus(_)))))) =>
+        ResponseClass.RetryableFailure
+    }
```

Comment:
I think we need to add a `_` case that leads to Non-retryable failure.  This covers the cases where the stream is empty and where the stream is a throw and where the stream doesn't have a grpc status.